clients/cli/src/unwrap.rs

```diff
@@ -0,0 +1,231 @@
+    #[clap(value_parser)]
+    pub amount: u64,
+
+    /// Signer source of transfer authority (to burn wrapped tokens)
+    /// if different from fee payer
+    #[clap(
+        long,
+        value_parser = SignerSourceParserBuilder::default().allow_all().build()
+    )]
+    pub transfer_authority: Option<SignerSource>,
+
+    /// The address of the unwrapped mint, queried if not provided
+    #[clap(long, value_parser = parse_pubkey)]
+    pub unwrapped_mint: Option<Pubkey>,
+
+    /// The address of the token program for the wrapped mint,
+    /// queried if not provided.
+    #[clap(long, value_parser = parse_token_program)]
+    pub wrapped_token_program: Option<Pubkey>,
+
+    /// The address of the token program for the unwrapped mint,
+    /// queried if not provided.
+    #[clap(long, value_parser = parse_token_program)]
+    pub unwrapped_token_program: Option<Pubkey>,
+
+    #[clap(long, value_parser = value_parser!(Hash))]
+    pub blockhash: Option<Hash>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UnwrapOutput {
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_mint_address: Pubkey,
+
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_token_program: Pubkey,
+
+    #[serde_as(as = "DisplayFromStr")]
+    pub recipient_token_account: Pubkey,
+
+    pub amount: u64,
+
+    pub signatures: Vec<Signature>,
+}
+
+impl Display for UnwrapOutput {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln_name_value(
+            f,
+            "Unwrapped token program:",
+            &self.unwrapped_token_program.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Unwrapped mint address:",
+            &self.unwrapped_mint_address.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Recipient unwrapped token account:",
+            &self.recipient_token_account.to_string(),
+        )?;
+        writeln_name_value(f, "Amount unwrapped:", &self.amount.to_string())?;
+
+        writeln!(f, "Signers:")?;
+        for signature in &self.signatures {
+            writeln!(f, "  {signature}")?;
+        }
+
+        Ok(())
+    }
+}
+
+impl QuietDisplay for UnwrapOutput {
+    fn write_str(&self, _: &mut dyn std::fmt::Write) -> std::fmt::Result {
+        Ok(())
+    }
+}
+impl VerboseDisplay for UnwrapOutput {}
+
+pub async fn command_unwrap(
+    config: &Config,
+    args: UnwrapArgs,
+    matches: &clap::ArgMatches,
+    wallet_manager: &mut Option<Rc<RemoteWalletManager>>,
+) -> CommandResult {
+    let payer = config.fee_payer()?;
+
+    // Get wrapped token program ID from the owner of the wrapped token account
+    // if not provided
+    let wrapped_token_program_id = if let Some(program_id) = args.wrapped_token_program {
+        program_id
+    } else {
+        get_account_owner(&config.rpc_client, &args.wrapped_token_account).await?
+    };
```

Comment:
For all of these args, if you're online (ie not in sign-only), it's good to check that the provided parameters are correct if they're explicitly provided.